modules/thermal_hydraulics/include/utils/Numerics.h

```diff
@@ -150,13 +170,18 @@ ADReal Peclet(ADReal volume_fraction, ADReal cp, ADReal rho, ADReal vel, ADReal
  *
  * @return Grashof number
  */
-Real Grashof(Real beta, Real dT, Real D_h, Real rho_liquid, Real mu_liquid, Real gravity_magnitude);
-ADReal Grashof(ADReal beta,
-               ADReal dT,
-               ADReal D_h,
-               ADReal rho_liquid,
-               ADReal mu_liquid,
-               Real gravity_magnitude);
+template <typename T>
+T
+Grashof(const T & beta,
+        const T & dT,
+        const T & D_h,
+        const T & rho_liquid,
+        const T & mu_liquid,
+        const T & gravity_magnitude)
```

Comment:
```suggestion
        const Real & gravity_magnitude)
```

Reply:
Gravity won't depend on Dofs, so just keep it Real. Do this for all the functions. That's the reason for at least some of the compile failures.